### DISCUSSION TITLE
About assigning node representation to graph data during training

### DISCUSSION THREAD

**Discussion by YiKangOY** [pyg-team/pytorch_geometric#6953] [Q&A]
I have two GNN models. The first will calculate the embedding of each node. 
I want to feed the graph data with the embedding calculated in GNN1 to GNN2, i.e., the embedding now becomes the graph.x for the graph sent to GNN2.

How can I do this? Maybe deepcopy the graphs sent to GNN1, assign the embeddings to them, and send them to GNN2?

**Comment by wsad1** [comment]
You can just pass the embeddings to GNN2 a `x`
```
out1 = GNN1(x = x, edge_index = edge_index)
out2 = GNN2(x = out1, edge_index = edge_index)
```
Also: if you don't want `GNN1` and `GNN2` to be trained end to end, call `GNN1.eval()`.

**Reply by YiKangOY** [reply]
Hi, GNN1 and GNN2 are consecutive, and I want to train them together. Since I will use not only x and edge_index in the convolution, what I want to do is:
out1 = GNN1(graph) #will use graph.x, graph.edge_index, and other graph features
graph.x = out1
out2 = GNN2(graph)#will use out 1 as the new graph.x, while keeping graph.edge_index and other graph features unchanged.

But this will destroy the computation graph of pytorch, so I want to use deepcopy to achieve it:
out1 = GNN1(graph) #will use graph.x, graph.edge_index, and other graph features
**new_graph = copy.deepcopy(graph)
new_graph.x = out1**
out2 = GNN2(new_graph)#will use out 1 as new_graph.x, while keeping new_graph.edge_index and other graph features unchanged.

I want to know if this is correct computationally and is there more efficient wat to achieve it.

**Reply by wsad1** [reply]
You don't need to do a deep copy
`graph.x = out1` will not effect the computation graph of pytroch.

**Reply by YiKangOY** [reply]
Hi, thanks for your reply. I will try it out and see if there are any other bugs in my code.
